src/compiler/utilities.ts

```diff
@@ -4344,8 +4342,20 @@ export function isPushOrUnshiftIdentifier(node: Identifier) {
     return node.escapedText === "push" || node.escapedText === "unshift";
 }
 
-/** @internal */
-export function isParameterDeclaration(node: VariableLikeDeclaration): boolean {
+// TODO(jakebailey): this function should not be named this. While it does technically
+// return true if the argument is a ParameterDeclaration, it also returns true for nodes
+// that are children of ParameterDeclarations inside binding elements.
+// Probably, this should be called `rootDeclarationIsParameter`.
+/**
+ * This function returns true if the this node's root declaration is a parameter.
+ * For example, passing a `ParameterDeclaration` will return true, as will passing a
+ * binding element that is a child of a `ParameterDeclaration`.
+ *
+ * If you are looking to test that a `Node` is a `ParameterDeclaration`, use `isParameter`.
+ *
+ * @internal
+ */
+export function isParameterDeclaration(node: Declaration): boolean {

```

Comment:
I agree with the comment. I have an aspirational goal of having better alignment between `SyntaxKind`, the names of each `Node` subtype, and the various node tests. It's constantly bugged me that we have `SyntaxKind.MethodDeclaration -> interface MethodDeclaration`, but `SyntaxKind.Constructor -> ConstructorDeclaration`.

I'd much rather we rename this to something like `isPartOfParameterDeclaration` or the like, and change `isParameter` to `isParameterDeclaration` (though `isParameter` is public, so we'd have to potentially deprecate it and forward it). I did something similar several years back with `isExpression -> isPartOfExpression` so that the simpler `isExpression` that exists now would be consistent with the naming of other node tests.

Reply:
Should I make this change in this PR?

Reply:
I'm not sure if we'll really ever be able to drop `isParameter`, though; doing code search, it seems pervasive (about 2.7k non-fork results for `ts.isParameter` on sourcegraph).